Orthtree/include/CGAL/Orthtree.h

```diff
@@ -538,77 +629,131 @@ class Orthtree
+
+  \param query query point
+  \param k number of neighbors to find
+  \param output output iterator
+
+  \warning Nearest neighbor searches requires `GeomTraits` to be a model of `CollectionPartitioningOrthtreeTraits`.
+ */
   template<typename OutputIterator>
-  OutputIterator nearest_neighbors (const Point& query,
-                                    std::size_t k,
-                                    OutputIterator output) const {
-    Sphere query_sphere (query, (std::numeric_limits<FT>::max)());
+  auto nearest_neighbors(const Point& query,
+    std::size_t k,
+    OutputIterator output) const -> std::enable_if_t<supports_neighbor_search, OutputIterator> {
+    Sphere query_sphere(query, (std::numeric_limits<FT>::max)());
+    CGAL_precondition(k > 0);
+
     return nearest_k_neighbors_in_radius(query_sphere, k, output);
   }
 
   /*!
-    \brief finds the points in `sphere`.
+  \brief finds the elements in the sphere `query`.
 
-    Nearest neighbors are outputted in order of increasing distance to
-    the center of `sphere`.
+  Elements are outputted in order of increasing distance to
+  the center of the sphere.
 
-    \tparam OutputIterator a model of `OutputIterator` that accept `Point_d` objects.
-    \param query query sphere.
-    \param output output iterator.
-   */
+  \tparam OutputIterator a model of `OutputIterator` that accepts `GeomTraits::Node_data_element` objects.
+
+  \param query query sphere
+  \param output output iterator
+
+  \warning Nearest neighbor searches requires `GeomTraits` to be a model of `CollectionPartitioningOrthtreeTraits`.
+ */
   template<typename OutputIterator>
-  OutputIterator nearest_neighbors (const Sphere& query, OutputIterator output) const {
+  auto neighbors_in_radius(const Sphere& query, OutputIterator output) const -> std::enable_if_t<supports_neighbor_search, OutputIterator> {
+    return nearest_k_neighbors_in_radius(query, (std::numeric_limits<std::size_t>::max)(), output);
+  }
+
+  /*!
+  \brief finds at most `k` elements within a specific radius that are
+  nearest to the center of the sphere `query`: if `query` does not contain
+  at least `k` elements, only contained elements will be returned.
+
+  This function is useful when the user already knows how sparse the elements are,
+  or if they do not care about elements that are too far away.
+  Setting a small radius may have performance benefits.
+
+  \tparam OutputIterator must be a model of `OutputIterator` that accepts `GeomTraits::Node_data_element`
+
+  \param query the region to search within
+  \param k the number of elements to find
+  \param output the output iterator to add the found elements to (in order of increasing distance)
+
+  \warning Nearest neighbor searches requires `GeomTraits` to be a model of `CollectionPartitioningOrthtreeTraits`.
+ */
+  template <typename OutputIterator>
+  auto nearest_k_neighbors_in_radius(
```

Comment:
Note that the function for points  has  no `_k_`  in the name. Shall we unify (adding or dropping)?

Reply:
I added `_k_`.